src/blant-utils.c

```diff
@@ -123,25 +123,30 @@ unsigned int L_K_Func(Gint_type Gint) {
 unsigned int L_K_Func(Gint_type Gint) {Apology("L_K_Func() not yet implemented"); return -1;}
 #endif
 
+Gordinal_type L_K_Smaller_Canon_Map(Gint_type Gint) {
+    unsigned char perm[MAX_K];
+    Gint_type canon_value = smaller_canon_map(Gint, _k, perm);
+    return canon_to_ordinal(canon_value, _k);
+}
+
 // Assuming the global variable _k is set properly, go read in and/or mmap the big global
 // arrays related to canonical mappings and permutations.
 void SetGlobalCanonMaps(void)
 {
     int i;
     char BUF[BUFSIZ];
-    assert(3 <= _k && _k <= 8);
+    assert(3 <= _k && _k <= 9);
 #if SELF_LOOPS
     _Bk = (1 <<(_k*(_k+1)/2));
 #else
-    _Bk = (1 <<(_k*(_k-1)/2));
+    _Bk = (1 <<(_k_base*(_k_base-1)/2));
```

Comment:
I think this needs to be the actual _k being passed on the command line. What happens here if we compile your code in, but the command-line asks for k<8, eg k=5? The _Bk should be using k=5, not k=8 (I assume _k_base is 8, yes?). Have you run the full regression test using your new code? Because that's what regression tests are FOR: to ensure new code doesn't break old functionality.

Reply:
No it's not 8
`_k_base = MIN(_k, OLD_MAX_K);`
So _k_base = _k for k <= 8